presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionProfiler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.gen;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import static com.google.common.base.Verify.verify;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class ExpressionProfiler
+{
+    private static final int NUMBER_OF_ROWS_TO_PROFILE = 1024;
```

Comment:
I'm concerned that the ramp up is going to be too slow, and then grow too large.  My understanding is we will do 1 row pages until we hit the profile size, and then we allow the batch size to increase based only on the output size.  This means if the operation takes 0.5 seconds, we will allow size to grow to max, but the the max runtime is `8 * 1024 * 0.5 seconds ~= 1.14 hours`.  Instead, I suggest we measure each batch and then simply say if the last batch was too large, since this fits the existing scaling model in `PageProjection` (we could say the exact size we expect to be correct, but I don't think we need that).  This means we will grow by doubles quickly, but still stop at an appropriate range and if the data changes we can continue to adjust.